lib/model/query/forms.js

```diff
@@ -1,10 +1,30 @@
+const { rowsToInstances, maybeRowToInstance } = require('../../util/db');
 
 module.exports = {
   create: (form) => ({ actees, simply }) =>
     actees.transacting
       .provision('form')
       .then((actee) => simply.create('forms', form.with({ acteeId: actee.id }))),
 
-  getAll: () => ({ simply, Form }) => simply.getAll('forms', Form)
+  getByXmlFormId: (xmlFormId, extended) => ({ forms, Form }) =>
+    forms._get(extended, { xmlFormId }).then(maybeRowToInstance(Form)),
+
+  getAll: (extended) => ({ forms, Form }) =>
+    forms._get(extended).then(rowsToInstances(Form)),
+
+  _get: (extended = false, condition = []) => ({ db, simply, Form }) => (extended === false)
+    ? simply.getAll('forms', Form)
```

Comment:
Should this also apply the `condition`, as well as exclude deleted forms?

Reply:
Aye. Fixed.